packages/assets-controllers/src/MultichainTokensRatesController/MultichainTokensRatesController.ts

```diff
@@ -0,0 +1,387 @@
+      await this.updateTokensRates(this.#accountId);
+    }
+  }
+
+  /**
+   * Determines whether the controller is active.
+   *
+   * @returns True if the keyring is unlocked; otherwise, false.
+   */
+  get isActive(): boolean {
+    return this.#isUnlocked;
+  }
+
+  /**
+   * Checks if an account is a non-EVM account with a Snap.
+   *
+   * @param account - The account to check.
+   * @returns True if the account is non-EVM and has Snap metadata; otherwise, false.
+   */
+  #isNonEvmAccount(account: InternalAccount): boolean {
+    return (
+      !isEvmAccountType(account.type) && account.metadata.snap !== undefined
+    );
+  }
+
+  /**
+   * Retrieves all multichain accounts from the AccountsController.
+   *
+   * @returns An array of internal accounts.
+   */
+  #listMultichainAccounts(): InternalAccount[] {
+    return this.messagingSystem.call(
+      'AccountsController:listMultichainAccounts',
+    );
+  }
+
+  /**
+   * Filters and returns non-EVM accounts that should have balances.
+   *
+   * @returns An array of non-EVM internal accounts.
+   */
+  #listAccounts(): InternalAccount[] {
+    const accounts = this.#listMultichainAccounts();
+    return accounts.filter((account) => this.#isNonEvmAccount(account));
+  }
+
+  /**
+   * Retrieves a non-EVM account by its ID.
+   *
+   * @param accountId - The account ID.
+   * @returns The corresponding internal account, or undefined if not found.
+   */
+  #getAccount(accountId: string): InternalAccount | undefined {
+    return this.#listAccounts().find((account) => account.id === accountId);
+  }
+
+  /**
+   * Handles the removal of an account by deleting its conversion rates.
+   *
+   * @param accountId - The ID of the removed account.
+   */
+  async #handleOnAccountRemoved(accountId: string): Promise<void> {
+    if (accountId in this.state.conversionRates) {
+      this.update((state: Draft<MultichainTokensRatesControllerState>) => {
+        delete state.conversionRates[accountId];
+      });
+    }
+  }
+
+  /**
+   * Updates the token conversion rates for a given account.
+   *
+   * This method acquires a mutex lock to ensure thread safety.
+   *
+   * @param accountId - The account ID for which to update rates.
+   * @returns A promise that resolves when the rates have been updated.
+   */
+  async updateTokensRates(accountId: string): Promise<void> {
+    const releaseLock = await this.#mutex.acquire();
+
+    return (async () => {
+      if (!this.isActive) {
+        return;
+      }
+
+      const account = this.#getAccount(accountId);
+
+      if (!account?.metadata.snap) {
+        return;
+      }
+
+      // Retrieve assets from the assets controller.
+      const assets = await this.#getAssetsList(
+        accountId,
+        account.metadata.snap.id as SnapId,
+      );
+
+      const accountType = account.type as KeyringAccountType;
+
+      const conversions = assets.map((asset) => ({
+        from: asset,
+        to: MAP_SWIFT_ISO4217[accountType],
```

Comment:
this will be whatever the value id set on the current selected currency by the user parsed as a caip

Reply:
I agree with @aganglada, we need a mapping for all the possible user's preferred currencies. This mapping doesn't mean anything since it doesn't follow [the standard](https://github.com/ChainAgnostic/namespaces/pull/132/files)

Reply:
To extend what @aganglada is saying I think we will have multiple possibilities here. It could be to ask a conversion from an asset to the users preferred currency or to ask a conversion from the native asset to another asset